src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -618,21 +619,22 @@ class TerminalTabsDragAndDrop implements IListDragAndDrop<ITerminalInstance> {
 		this._autoFocusInstance = undefined;
 
 		let sourceInstances: ITerminalInstance[] | undefined;
-		const terminalResources = originalEvent.dataTransfer?.getData(DataTransfers.RESOURCES);
-		const terminals = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS);
 		let promises: Promise<IProcessDetails | undefined>[] = [];
-		if (terminals && terminalResources) {
-			const json = JSON.parse(terminalResources);
+		const resources = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS) || originalEvent.dataTransfer?.getData(DataTransfers.RESOURCES);
```

Comment:
Can we just do the TERMINALS one now that we have the resource change done? Ideally we wouldn't hit the `if (terminalIdentifier.instanceId) {` if it's a file for example.